File: src/design_patterns/structural/facade/c#/facade.cs
class SubsystemA
{
    public string OperationA()
    {
        return "Subsystem A operation";
    }
}

class SubsystemB
{
    public string OperationB()
    {
        return "Subsystem B operation";
    }
}

class Facade
{
    private readonly SubsystemA _subsystemA;
    private readonly SubsystemB _subsystemB;

    public Facade(SubsystemA subsystemA, SubsystemB subsystemB)
    {
        _subsystemA = subsystemA;
        _subsystemB = subsystemB;
    }

    public string Operation()
    {
        var results = new List<string>();
        results.Add("Facade initializes subsystems:");
        results.Add(_subsystemA.OperationA());
        results.Add(_subsystemB.OperationB());
        return string.Join("\n", results);
    }
}

/*
// Usage
var subsystemA = new SubsystemA();
var subsystemB = new SubsystemB();
var facade = new Facade(subsystemA, subsystemB);
Console.WriteLine(facade.Operation());  // Output: Facade initializes subsystems:\nSubsystem A operation\nSubsystem B operation
*/

File: src/design_patterns/structural/facade/javascript/facade.js
class SubsystemA {
    operationA() {
      return "Subsystem A operation";
    }
  }
  
  class SubsystemB {
    operationB() {
      return "Subsystem B operation";
    }
  }
  
  class Facade {
    constructor(subsystemA, subsystemB) {
      this._subsystemA = subsystemA;
      this._subsystemB = subsystemB;
    }
  
    operation() {
      let results = [];
      results.push("Facade initializes subsystems:");
      results.push(this._subsystemA.operationA());
      results.push(this._subsystemB.operationB());
      return results.join("\n");
    }
  }
  
  // Usage
  const subsystemA = new SubsystemA();
  const subsystemB = new SubsystemB();
  const facade = new Facade(subsystemA, subsystemB);
  console.log(facade.operation());  // Output: Facade initializes subsystems:\nSubsystem A operation\nSubsystem B operation
  